### PR TITLE
Support for Labeling and Tolerations were added

### PR DESCRIPTION
**Pull request by p-pekala**
Support for lables and tolerations in k8s

### CODE REVIEW

File: kedro_airflow_k8s/operators/node_pod.py [hunk 1]
```diff
@@ -96,6 +98,9 @@ def __init__(  # pylint: disable=too-many-arguments
             is_delete_operator_pod=True,
             pod_template_file=self.minimal_pod_template,
             node_selector=node_selector_labels,
+            node_selectors=node_selector_labels,
```
Comment (michalzelechowski-getindata) [inline kedro_airflow_k8s/operators/node_pod.py:101]: Is the repetition to line 100 necessary? It is assigned to the same k8s operator attribute in the end.

File: kedro_airflow_k8s/operators/node_pod.py [hunk 2]
```diff
@@ -37,6 +37,8 @@ def __init__(  # pylint: disable=too-many-arguments
         limits_cpu: Optional[str] = None,
         limits_memory: Optional[str] = None,
         node_selector_labels: Optional[Dict[str, str]] = None,
+        labels: Optional[Dict[str, str]] = None,
+        tolerations: Optional[List[Dict[str, str]]] = None,
```
Comment (michalzelechowski-getindata) [inline kedro_airflow_k8s/operators/node_pod.py:41]: These parameters are missing in function doc section.

File: kedro_airflow_k8s/config.py [hunk 3]
```diff
@@ -132,6 +132,15 @@ def memory(self):
 
 
 class ResourceConfig(Config):
+
+    @property
+    def tolerations(self):
```
Comment (michalzelechowski-getindata) [inline kedro_airflow_k8s/config.py:137]: Add tolerations and labels to DEFAULT_CONFIG_TEMPLATE as it's used by init task.

**Review by michalzelechowski-getindata — CHANGES_REQUESTED** [review]
Good work! Keep in mind to put commenting line into CHANGELOG.md

File: kedro_airflow_k8s/operators/node_pod.py (hunk 1 above)
Comment (ppekala-ias) [reply kedro_airflow_k8s/operators/node_pod.py:101]: fixed

File: kedro_airflow_k8s/operators/node_pod.py (hunk 2 above)
Comment (ppekala-ias) [reply kedro_airflow_k8s/operators/node_pod.py:41]: fixed

File: kedro_airflow_k8s/config.py (hunk 3 above)
Comment (ppekala-ias) [reply kedro_airflow_k8s/config.py:137]: fixed

File: kedro_airflow_k8s/operators/node_pod.py [hunk 4]
```diff
@@ -174,3 +179,23 @@ def create_security_context(
...
+        :return:
+        """
+        if not tolerations:
+            return []
+        return [
+            k8s.V1Toleration(
+                effect=toleration.get("effect"),
+                key=toleration.get("key"),
+                operator=toleration.get("operator"),
+                value=toleration.get("value"))
```
Comment (michalzelechowski-getindata) [inline kedro_airflow_k8s/operators/node_pod.py:199]: Are all tolerations attributes required or some of these are optional? If some are optional, use `toleration.get("operator",None) instead.

Comment (ppekala-ias) [reply kedro_airflow_k8s/operators/node_pod.py]: I think is should be ok:

>>> test = {'xxx': 'test'}
>>> test.get('xxx')
'test'
>>> test.get('yyy')
>>> print(test.get('yyy'))
None
>>> type(test.get('yyy'))
<class 'NoneType'>

**Review by michalzelechowski-getindata — APPROVED** [review]
LGTM

**Pull request merged**